### PR TITLE
redis: don't include storage_proxy.hh in commands.cc

### PR DESCRIPTION
**Pull request by kbr-**
Use a forward declaration instead.

Tracker issue: https://github.com/scylladb/scylla/issues/6523.

### CODE REVIEW

File: redis/commands.cc [hunk 1]
```diff
@@ -25,13 +25,16 @@
...
 #include "service_permit.hh"
-#include "service/storage_proxy.hh"
 #include "service/client_state.hh"
 #include "redis/options.hh"
 #include "redis/query_utils.hh"
 #include "redis/mutation_utils.hh"
 #include "redis/lolwut.hh"
 
+namespace service {
+    class storage_proxy;
```
Comment (tgrabiec) [inline redis/commands.cc:35]: Style: We don't indent inside namespace block.

**Comment by nyh** [comment]
Kamil, most of the header files in redis/ already have "class storage_proxy;". Can you please check if commands.cc really doesn't include those and requires doing this too? Maybe simply dropping the include of storage_proxy.hh can be enough?

**Comment by kbr-** [comment]
@nyh thanks for pointing that out. There was an even bigger problem: commands.cc indirectly includes abstract_command.hh, which also included storage_proxy.hh. So this PR actually didn't do anything.
Now it's better.

File: redis/commands.cc (hunk 1 above)
Comment (kbr-) [reply redis/commands.cc]: Now the forward declaration is somewhere else, this time uses the proper style.

**Comment by psarna** [comment]
queued as a304f774f8fb7ba3013ef9ac489fbfc0b4080a69. Thanks for the clean-up!

**Comment by bhalevy** [comment]
@kbr- a304f77 can close this PR, but you mentioned:
> @nyh thanks for pointing that out. There was an even bigger problem: commands.cc indirectly includes abstract_command.hh, which also included storage_proxy.hh. So this PR actually didn't do anything.

Are you going to open a new PR?

**Comment by kbr-** [comment]
No, I fixed the mentioned issue in this PR. Closing.

**Pull request closed without merging**